src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -2,214 +2,151 @@
-        //          This should be based off some sort of "known headers/values" scheme, so we don't need
-        //          to re-lookup header names each time.
         // * Huffman encoding
         //      
         // Things we should consider adding:
-        // * Dynamic table encoding.
-        //          This would make the encoder stateful, which complicates things significantly.
-        //          Additionally, it's not clear exactly what strings we would add to the dynamic table
-        //          without some additional guidance from the user about this.
-        //          So for now, don't do dynamic encoding.
+        // * Dynamic table encoding:
+        //   This would make the encoder stateful, which complicates things significantly.
+        //   Additionally, it's not clear exactly what strings we would add to the dynamic table
+        //   without some additional guidance from the user about this.
+        //   So for now, don't do dynamic encoding.
 
-        public static bool EncodeIndexedField(int index, Span<byte> buffer, out int length)
+        public static bool EncodeIndexedField(int index, Span<byte> destination, out int bytesWritten)
         {
-            length = 0;
-
-            if (buffer.Length == 0)
-            {
-                return false;
-            }
-
-            buffer[0] = 0x80;   // Literal field
-            if (!IntegerEncoder.Encode(index, 7, buffer, out length))
+            if (destination.Length != 0)
             {
-                return false;
+                destination[0] = 0x80;   // Literal field
+                return IntegerEncoder.Encode(index, 7, destination, out bytesWritten);
             }
 
-            return true;
+            bytesWritten = 0;
+            return false;
         }
 
-        public static bool EncodeIndexedName(int index, string value, Span<byte> buffer, out int length)
+        public static bool EncodeIndexedName(int index, string value, Span<byte> destination, out int bytesWritten)
         {
-            int i = 0;
-            length = 0;
-
-            if (buffer.Length == 0)
+            if (destination.Length != 0)
             {
-                return false;
-            }
-
-            buffer[0] = 0x40;   // Literal name
-            if (!IntegerEncoder.Encode(index, 6, buffer, out int indexLength))
-            {
-                return false;
-            }
-
-            i = indexLength;
-            if (i == buffer.Length)
-            {
-                return false;
-            }
-
-            if (!EncodeString(value, buffer.Slice(i), out int nameLength, lowercase: false))
-            {
-                return false;
+                destination[0] = 0x40;   // Literal name
+                if (IntegerEncoder.Encode(index, 6, destination, out int indexLength))
+                {
+                    if (indexLength < destination.Length &&
```

Comment:
I don't think you need the length check here. EncodeString is also doing the check. 

There are several places below where we are checking before calling EncodeString, and some where we aren't. Seems like we should be consistent about not checking and just let EncodeString do the check.

Reply:
I don't think it's needed either; I was just carrying forward the checks already there. I'll remove them.